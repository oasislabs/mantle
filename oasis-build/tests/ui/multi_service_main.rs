use oasis_std::Context;

#[derive(oasis_std::Service)]
pub struct Counter(u32);

#[derive(oasis_std::Service)]
pub struct Counter2(u32);

impl Counter {
    pub fn new(ctx: &Context) -> Result<Self, ()> {
        Ok(Self(42))
    }
}

impl Counter2 {
    pub fn new(ctx: &Context) -> Result<Self, ()> {
        Ok(Self(42))
    }
}

fn main() {
    oasis_std::service!(Counter);
    oasis_std::service!(Counter2);
}
