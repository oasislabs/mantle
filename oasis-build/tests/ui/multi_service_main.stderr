error: Multiple invocations of `oasis_std::service!`. Second occurrence here:
  --> $DIR/multi_service_main.rs:23:5
   |
23 |     oasis_std::service!(Counter2);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

